lib/worker/worker.js

```diff
@@ -0,0 +1,210 @@
+  const { eslintLint } = meta;
+  return lintPathOrContent(eslintLint, filePath, fileContent);
+}
+
+async function jobESLintLint ({
+  projectPath, config, filePath, isModified, contents
+}) {
+  const eslint = getESLint(filePath, config, { projectPath });
+  process.chdir(eslint.cwd);
+
+  const lintResults = await lintFile(eslint, filePath, contents);
+  const rulesMeta = eslint.eslintLint.getRulesMetaForResults(lintResults);
+
+  return {
+    rules: rulesMeta,
+    results: convertResults(
+      lintResults,
+      rulesMeta,
+      config,
+      { isModified }
+    ),
+  };
+}
+
+async function fixFile (meta, filePath, fileContent) {
+  const { ESLint, eslintFix } = meta;
+  const results = await lintPathOrContent(eslintFix, filePath, fileContent);
+  await ESLint.outputFixes(results);
+  return results;
+}
+
+function countMessages (results) {
+  let count = 0;
+  for (let { messages } of results) {
+    count += messages.length;
+  }
+  return count;
+}
+
+async function jobESLintFix ({
+  projectPath, config, filePath, isModified, contents
+}) {
+  const eslint = getESLint(filePath, config, { projectPath });
+  process.chdir(eslint.cwd);
+
+  const lintResults = await lintFile(eslint, filePath, contents);
+  const lintMessageCount = countMessages(lintResults);
+
+  const fixResults = await fixFile(eslint, filePath, contents);
+  const rulesMeta = eslint.eslintFix.getRulesMetaForResults(fixResults);
+
+  const result = {
+    rules: rulesMeta,
+    results: convertResults(
+      fixResults,
+      rulesMeta,
+      config,
+      { isModified }
+    ),
+  };
+
+  if (lintMessageCount > 0) {
+    result.fixCount = lintMessageCount - result.results.length;
+  }
+
+  return result;
+}
+
+async function processMessage (bundle) {
+  const oldCwd = process.cwd();
+
+  try {
+    const { type } = bundle;
+    if (type == null) {
+      throw new JobTypeError();
+    }
+
+    switch (type) {
+      case 'clear-cache': return jobClearCache(bundle);
+      case 'debug': return jobDebug(bundle);
+      case 'lint': return jobESLintLint(bundle);
+      case 'fix': return jobESLintFix(bundle);
+      default: throw new JobNotFoundError(type);
+    }
+  } finally {
+    process.chdir(oldCwd);
+  }
+}
+
+function handleError(error, { key, filePath, projectPath } = {}) {
+  if (error.messageTemplate === 'no-config-found') {
+    error = new ConfigNotFoundError(error.message, filePath, projectPath);
+  }
+
+  if (error instanceof Error) {
+    error.error = error.message;
+    error.key = key;
+
+    return error;
+  }
+
+  return new UnknownError();
+}
+
+if (require.main === module) {
+  process.stdin
+  .pipe(ndjson.parse({ strict: false }))
+  .on('data', (data) => {
+    Messaging.runWith(data, async () => {
+      try {
+        if (!data.key) {
+          throw new JobKeyError();
+        }
+
+        Messaging.emit(await processMessage(data));
+      } catch (error) {
+        Messaging.emitError(handleError(error, data));
+      }
+    });
```

Comment:
This is now effectivly the only place to respond to requests. It allows for more consise  response handling.

All jobs now just return what the job wants to output, or throws an error.